### PR TITLE
[CI] MacOS 15 testing

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Adds MacOS 15 ARM tests unit tests for PRs and merge to release branches.
Makes MacOS 13 x86 tests run only on main and release branches

MacOS 14 is expected in a follow-up PR
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/4038

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -84,26 +85,27 @@ steps:
...
+          imagePrefix: generic-base-15-arm-002
         retry:
           manual:
             allowed: true
 
+      # Runs only on main and release branches
       - label: "Unit tests - MacOS 13"
-        key: "unit-tests-macos-13"
         command: ".buildkite/scripts/steps/unit-tests.sh"
+        branches: "main 8.x 8.16 9.*"
```
Comment (rowlandgeoff) [inline .buildkite/pipeline.yml:108]: How about 8.* instead? Should prevent us from having to update the branch every release.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -84,26 +85,27 @@ steps:
...
-        branches: main
         artifact_paths:
           - "build/TEST-*.html"
           - "build/TEST-*.xml"
           - "build/diagnostics/*"
           - "coverage-*.out"
         agents:
           provider: orka
-          imagePrefix: generic-13-ventura-arm
+          imagePrefix: generic-base-15-arm-002
```
Comment (rowlandgeoff) [inline .buildkite/pipeline.yml:100]: What's with the `002`? That comes across as a somewhat arbitrary naming scheme.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -84,26 +85,27 @@ steps:
           manual:
             allowed: true
 
-      # Runs inly on the main branch
-      - label: "Unit tests - MacOS 13 ARM"
-        key: "unit-tests-macos-13-arm"
+  - group: "MacOS tests"
```
Comment (rowlandgeoff) [inline .buildkite/pipeline.yml:88]: The official naming is now `macOS`, which we adopted in the beats repo. Might as well update it here too.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (pazone) [reply .buildkite/pipeline.yml:100]: We have an agreement for Orka images versioning. When we build a new image is built we increment its suffix to make sure we don't affect any pipelines

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (pazone) [reply .buildkite/pipeline.yml]: Renamed

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (pazone) [reply .buildkite/pipeline.yml]: Good idea!

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -84,26 +85,27 @@ steps:
           manual:
             allowed: true
 
-      # Runs inly on the main branch
-      - label: "Unit tests - MacOS 13 ARM"
-        key: "unit-tests-macos-13-arm"
+  - group: "macOS tests"
+    key: "macos-unit-tests"
+    steps:
+      - label: "Unit tests - MacOS 15 ARM"
```
Comment (rowlandgeoff) [inline .buildkite/pipeline.yml:91]:
```suggestion
      - label: "Unit tests - macOS 15 ARM"
```

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -84,26 +85,27 @@ steps:
...
         agents:
           provider: orka
-          imagePrefix: generic-13-ventura-arm
+          imagePrefix: generic-base-15-arm-002
         retry:
           manual:
             allowed: true
 
+      # Runs only on main and release branches
       - label: "Unit tests - MacOS 13"
```
Comment (rowlandgeoff) [inline .buildkite/pipeline.yml:106]:
```suggestion
      - label: "Unit tests - macOS 13"
```

**Review by rowlandgeoff — APPROVED** [review]
Looks good, just those couple of nit-picks

**Pull request merged**